src/sentry/models/activity.py

```diff
@@ -36,20 +36,24 @@ def get_activities_for_group(self, group: Group, num: int) -> Sequence[Group]:
         activity_qs = self.filter(group=group).order_by("-datetime")
 
         if not features.has("projects:issue-priority", group.project):
-            activity_qs.exclude(type=ActivityType.SET_PRIORITY.value)
+            activity_qs = activity_qs.exclude(type=ActivityType.SET_PRIORITY.value)
 
         prev_sig = None
         sig = None
         # we select excess so we can filter dupes
         for item in activity_qs[: num * 2]:
             prev_sig = sig
-            sig = (item.type, item.ident, item.user_id)
+            sig = (item.type, item.ident, item.user_id, item.data)
 
             if item.type == ActivityType.NOTE.value:
                 activities.append(item)
                 continue
 
             if sig != prev_sig:
+                if item.type == ActivityType.SET_PRIORITY.value and (
+                    not item.data or "priority" not in item.data
+                ):
+                    continue
```

Comment:
Just curious what cases cause this? Do we just have bad historical data?

Reply:
Also curious why this was necessary - I don't think it should be possible

Reply:
I just put it in to be extra careful. When I was writing the tests I could have manually mocked the activity with data=None. Although this wouldn't reflect the reality of how that data would be populated, I just thew it in just in case. If you think that we will not be able to get to this point in the code with bad data, I can remove it.